astraceroute.c

```diff
@@ -4,7 +4,9 @@
  * Subject to the GPL, version 2.
  */
 
-#define _BSD_SOURCE
+#ifdef _BSD_SOURCE
+# define _DEFAULT_SOURCE
+#endif
```

Comment:
Judging from the documentation at http://man7.org/linux/man-pages/man7/feature_test_macros.7.html it seems we should define both, _BSD_SOURCE and _DEFAULT regardless of any existing definitions, i.e.

`#define _BSD_SOURCE`
`#define _DEFAULT_SOURCE`

Does this fix the warnings as well (sorry, I don't have a system with glibc >= 2.20 available right now)? If yes, could you please update this patch (just force push the updated commit to your branch). Thanks!